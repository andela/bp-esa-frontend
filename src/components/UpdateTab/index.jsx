import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const UpdateTab = ({ numberOfItems, handleUpdates, view }) => (
  <div>
    <button type="button" className={`update-tab ${view === 'cardView' && 'card-width'}`} onClick={handleUpdates}>
    Load new updates &nbsp;[
      {numberOfItems}
]
    </button>
  </div>
);

UpdateTab.propTypes = {
  numberOfItems: PropTypes.number.isRequired,
  handleUpdates: PropTypes.func.isRequired,
  view: PropTypes.string.isRequired,
};

export default UpdateTab;
