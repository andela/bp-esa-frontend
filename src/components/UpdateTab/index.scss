@-webkit-keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.update-tab{
  border: none;
  border-radius: 5px;
  margin-top: 20px;
  text-align: center;
  width: 100%;
  padding: 15px;
  background: lighten($color: #385de0, $amount: 40%);
  cursor: pointer;
  color: lighten($color: #385de0, $amount: 10%);
  transition: background-color, 0.2s;
  -webkit-animation: fade-in 0.3s cubic-bezier(0.550, 0.055, 0.675, 0.190) both;
	animation: fade-in 0.5s cubic-bezier(0.550, 0.055, 0.675, 0.190) both;

  &:hover{
    background-color: lighten($color: #385de0, $amount: 35%);
  }
  &:focus{
    outline: none;
  }
}

.card-width{
  margin: 4px 80px 10px 80px;
  width: 88.8% !important;
}
