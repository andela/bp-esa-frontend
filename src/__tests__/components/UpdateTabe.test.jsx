import React from 'react';
import { shallow } from 'enzyme';
import UpdateTab from '../../components/UpdateTab';

describe('Update Tab', () => {
  const props = {
    numberOfItems: 1,
    handleUpdates: jest.fn(),
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<UpdateTab {...props} />);
  });

  it('should render component successfully', () => {
    expect(wrapper.find('.update-tab')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the handleUpdates method', () => {
    wrapper.find('.update-tab').simulate('click');
    expect(props.handleUpdates).toHaveBeenCalled();
  });
});
