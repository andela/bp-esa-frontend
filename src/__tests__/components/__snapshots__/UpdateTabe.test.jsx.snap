// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Tab should render component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UpdateTab
    handleUpdates={[MockFunction]}
    numberOfItems={1}
    view="listView"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="update-tab false"
        onClick={[MockFunction]}
        type="button"
      >
        Load new updates  [
        1
        ]
      </button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Load new updates  [",
          1,
          "]",
        ],
        "className": "update-tab false",
        "onClick": [MockFunction],
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        "Load new updates  [",
        1,
        "]",
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="update-tab false"
          onClick={[MockFunction]}
          type="button"
        >
          Load new updates  [
          1
          ]
        </button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Load new updates  [",
            1,
            "]",
          ],
          "className": "update-tab false",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          "Load new updates  [",
          1,
          "]",
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
